api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -52,8 +60,30 @@ public void setAppointmentAuditDao(AppointmentAuditDao appointmentAuditDao) {
         this.appointmentAuditDao = appointmentAuditDao;
     }
 
+    private boolean validateIfUserHasSelfOrAllAppointmentsAccess(Appointment appointment) {
+        return Context.hasPrivilege(MANAGE_APPOINTMENTS) ||
+                isAppointmentNotAssignedToAnyProvider(appointment) ||
+                isCurrentUserSamePersonAsOneOfTheAppointmentProviders(appointment.getProviders());
+    }
+
+    private boolean isAppointmentNotAssignedToAnyProvider(Appointment appointment) {
+        return isNull(appointment.getProviders()) ||
```

Comment:
All these can be pushed to the appointment.getProvidersWithResponse() - right? which will make the "isAppointmentNotAssignedToAnyProvider()" redundant?